src/geometry/euclidean/circumcenter.lean

```diff
@@ -378,31 +378,67 @@ begin
           (λ i, hr i (set.mem_univ _))).symm
 end
 
+/-- If there exists a distance that a point has from all vertices of a
+simplex, the orthogonal projection of that point onto the subspace
+spanned by that simplex is its circumcenter.  Version with distance
+condition using `set.range`. -/
+lemma orthogonal_projection_eq_circumcenter_of_exists_dist_eq {n : ℕ} (s : simplex ℝ P n)
+  {p : P} (hr : ∃ r, ∀ p₁ ∈ set.range s.points, dist p₁ p = r) :
```

Comment:
Is this `set.range` version used often? I would expect the version with the `'` to be the "default".

Reply:
It turns out I was only using the `'` versions, so I've eliminated the other versions. I think the original motivation for splitting it up was that the first step of the proof uses `exists_dist_eq_iff_exists_dist_orthogonal_projection_eq`, which requires a set of points, so now the first step is rewriting backwards with `set.forall_range_iff` (with a forwards rewrite later in the proof) - and for some reason that backwards rewrite only works with the `change` / `conv` combination I now have, not with a plain `simp_rw`. But in terms of users of the result, rather than structure of the proof, the version without `set.range` seems better.